### PR TITLE
Rename `moment` to `support_point`

### LINKED ISSUE #6974: Rename `moment` to `finite_logp_point`
### Description

The name `moment` has raised confusion a couple of times, recently in #6804 

What that function is supposed to do is to return a point at which the distribution logp is finite (if any point exists).
This is often the mean or mode, but need not always be. We should rename it accordingly

### PR DESCRIPTION
**Pull request by aerubanov**
## Description
Rename `moment` method to avoid confusion with distribution mean. 

## Related Issue
- [x] Closes #6974 

## Checklist
- [x] Rename `moment` method in code (and other related variable names)
- [X] Update documentation

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [x] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc--7166.org.readthedocs.build/en/7166/

### CODE REVIEW

**Comment by aerubanov** [comment]
I rename `moment` to `finite_logp_point` in code. Will change docs accordingly next step.

**Comment by aerubanov** [comment]
@ricardoV94  could you please take a look?

**Comment by ricardoV94** [comment]
@aerubanov great work. I have some suggestions, let me know what you think:
1. Instead of `finite_logp_point`, call it `support_point`? A bit less verbose and still precise
2. Don't break code for people defining Distributions with `moment` methods or using the old `_moment` dispatch or the user-helper `moment`. Instead issue an informative FutureWarning about the name change.

**Comment by aerubanov** [comment]
@ricardoV94 your suggestions sounds good, will implement it

File: pymc/testing.py [hunk 1]
```diff
@@ -653,34 +653,34 @@ def check_selfconsistency_discrete_logcdf(
             )
 
 
-def assert_moment_is_expected(model, expected, check_finite_logp=True):
+def assert_support_point_is_expected(model, expected, check_finite_logp=True):
```
Comment (ricardoV94) [inline pymc/testing.py:665]: This is also user facing (in that we use this utility in other libraries), so should also add a wrapper with a deprecation warning.

Comment (aerubanov) [reply pymc/testing.py:665]: I added option for using old name

**Review by ricardoV94 — APPROVED** [review]

**Pull request merged**